patches/net/minecraft/world/item/BowItem.java.patch

```diff
@@ -1,14 +1,24 @@
 --- a/net/minecraft/world/item/BowItem.java
 +++ b/net/minecraft/world/item/BowItem.java
-@@ -26,7 +_,7 @@
-     public void releaseUsing(ItemStack p_40667_, Level p_40668_, LivingEntity p_40669_, int p_40670_) {
-         if (p_40669_ instanceof Player player) {
+@@ -28,13 +_,15 @@
              ItemStack itemstack = player.getProjectile(p_40667_);
--            if (!itemstack.isEmpty()) {
-+            if (!itemstack.isEmpty()) { // Porting 1.20.5 redo EventHooks.onArrowLoose
+             if (!itemstack.isEmpty()) {
                  int i = this.getUseDuration(p_40667_, p_40669_) - p_40670_;
++                i = net.neoforged.neoforge.event.EventHooks.onArrowLoose(itemstack, p_40668_, player, i, !itemstack.isEmpty());
++                if (i < 0) return;
                  float f = getPowerForTime(i);
                  if (!((double)f < 0.1)) {
+                     List<ItemStack> list = draw(p_40667_, itemstack, player);
+-                    if (p_40668_ instanceof ServerLevel serverlevel && !list.isEmpty()) {
+-                        this.shoot(serverlevel, player, player.getUsedItemHand(), p_40667_, list, f * 3.0F, 1.0F, f == 1.0F, null);
++                    if (p_40668_ instanceof ServerLevel serverLevel && !list.isEmpty()) {
++                        this.shoot(serverLevel, player, player.getUsedItemHand(), p_40667_, list, f * 3.0F, 1.0F, f == 1.0F, null);
++
```

Comment:
Revert the change to the `ServerLevel` variable name (to the original `serverlevel`; note the capitalization of the middle L).